telegram/ext/_applicationbuilder.py

```diff
@@ -564,7 +565,7 @@ def pool_timeout(self: BuilderType, pool_timeout: Optional[float]) -> BuilderTyp
         self._pool_timeout = pool_timeout
         return self
 
-    def http_version(self: BuilderType, http_version: str) -> BuilderType:
+    def http_version(self: BuilderType, http_version: Literal["1.1", "2"]) -> BuilderType:
```

Comment:
I would suggest doing something like
```py
HTTPVersion = Literal["1.1", "2"]
...
def http_version(self: BuilderType, http_version: HTTPVersion)
```
since it's repeated at least twice, but I'm not sure whether mypy will complain or not.

I think, BTW, that PyCharm still has poor support for literals, unless that changed in the newest version. Like you iterate over strings that clearly conform with the limitations of a given literal, and it still complains that you pass in a `str` when a `Literal` is expected.